File: AppGUI/coregui/mainwindow/mainwindow.cpp
#include "mainwindow.h"
#include "fancytabwidget.h"
#include "manhattanstyle.h"
#include "taskselectorwidget.h"
#include "welcomemanager.h"
#include "samplemanager.h"
#include "experimentmanager.h"
#include "simulationmanager.h"
#include "fitmanager.h"
#include "stylehelper.h"
#include "ba_stylehelper.h"

#include <QApplication>
#include <iostream>
#include <QStatusBar>

MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent)
//    : FancyMainWindow(parent)
    , m_tabWidget(0)
    , m_welcomeManager(0)
    , m_sampleManager(0)
    , m_expManager(0)
    , m_simulManager(0)
    , m_fitManager(0)
{
    QString baseName = QApplication::style()->objectName();
    qApp->setStyle(new ManhattanStyle(baseName));

//    Manhattan::Utils::StyleHelper::setBaseColor(QColor(Manhattan::Utils::StyleHelper::DEFAULT_BASE_COLOR));
    Manhattan::Utils::StyleHelper::setBaseColor(QColor(0xCC6600));

//    BornAgain::Utils::StyleHelper::setBaseColor(QColor(0xCC6600));


    setDockNestingEnabled(true);

    setCorner(Qt::BottomLeftCorner, Qt::LeftDockWidgetArea);
    setCorner(Qt::BottomRightCorner, Qt::BottomDockWidgetArea);

    //m_tabWidget = new TaskSelectorWidget(this);
    m_tabWidget = new Manhattan::FancyTabWidget(this);
    m_welcomeManager = new WelcomeManager();
    m_sampleManager = new SampleManager();
    m_expManager = new ExperimentManager();
    m_simulManager = new SimulationManager();
    m_fitManager = new FitManager();

    m_tabWidget->insertTab(0, m_welcomeManager, QIcon("./images/mode_welcome.png"), "Welcome");
    m_tabWidget->insertTab(1, m_sampleManager, QIcon("./images/mode_sample.png"), "Sample");
    m_tabWidget->insertTab(2, m_expManager, QIcon("./images/mode_exp.png"), "Experiment");
    m_tabWidget->insertTab(3, m_simulManager, QIcon("./images/mode_simul.png"), "Simulation");
    m_tabWidget->insertTab(4, m_fitManager, QIcon("./images/mode_fit.png"), "Fit");

    setCentralWidget(m_tabWidget);

//    m_tabWidget->statusBar()->setProperty("p_styled", true);
    setAcceptDrops(true);

}

MainWindow::~MainWindow()
{

}

File: GUI/pygui/mainwindow/MainWindow.py
import sys
import os
import sip

from PyQt4.QtCore import (
    Qt, QObject, QString
)

from PyQt4.QtGui import (
    QApplication, qApp, QMainWindow, QWidget, QIcon, QColor, QStyle
)

sys.path.append(os.path.abspath("../../lib"))
sys.path.append(os.path.abspath(".."))


def toPyQt(object, type):
    ptr = object.unwrap()
    return sip.wrapinstance(ptr, type)

def fromPyQt(object, type):
    ptr = sip.unwrapinstance(object)
    return type.wrap(ptr)


#import ManhattanStyle as lms
import libManhattanStyle as lms
from views.WelcomeView import WelcomeView

class MainWindow(QMainWindow):
    def __init__(self, parent=None):
        QMainWindow.__init__(self, parent)

        baseName = QApplication.style().objectName()

        #manhattan_style = lms.ManhattanStyle( str(baseName) )
        #manhattan_style = lms.ManhattanStyle( sip.unwrapinstance(baseName) )
        #manhattan_style = lms.ManhattanStyle( fromPyQt(baseName, lms.QString) )

        #lms.StyleHelper.setBaseColor( fromPyQt(QColor(0x086FA1), lms.QColor) )

        #k = QColor(Qt.red)
        #lms.StyleHelper.setBaseColor(k )
        #qApp.setStyle( lms.toPyQt(manhattan_style) )

        self.__tabWidget = lms.FancyTabWidget(self)
        self.__welcomeView = WelcomeView()

        print "PPPPP 1.1"
        self.__tabWidget.insertTab(0, self.__welcomeView,  QIcon("images/mode_welcome.png"), "Welcome" )
        print "PPPPP 1.2"
        #self.__tabWidget.insertTab(0, fromPyQt(self.__welcomeView, lms.QWidget), fromPyQt(QIcon(":/images/mode_welcome.png"), lms.QIcon), fromPyQt(QString("Welcome"), lms.QString ))

        self.setCentralWidget( lms.toPyQt(self.__tabWidget) )
        #self.setCentralWidget( self.__welcomeView )



